Add goto-harness man page

tautschnig opened the pull request:
This was built using help2man and then manually expanded. In the process
of preparing it, several errors in existing goto-harness documentation
were found and fixed as well.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Add goto-harness man page

This was built using help2man and then manually expanded. In the process
of preparing it, several errors in existing goto-harness documentation
were found and fixed as well.

peterschrammel commented on src/goto-harness/memory_snapshot_harness_generator.cpp:
```diff
@@ -111,7 +111,7 @@ void memory_snapshot_harness_generatort::validate_options(
     throw invalid_command_line_argument_exceptiont(
       "option --memory_snapshot is required",
-      "--harness-type initialise-from-memory-snapshot");
+      "--harness-type initialise-with-memory-snapshot");
```
`initialise-with-memory-snapshot` -> `initialize-with-memory-snapshot` (to be consistent with AE spelling we use everywhere else) 🇺🇸

tautschnig replied:
I've added a separate commit fixing this across code, documentation, tests.

peterschrammel commented on src/goto-harness/memory_snapshot_harness_generator_options.h:
```diff
@@ -33,7 +33,7 @@ Author: Diffblue Ltd.
 #define MEMORY_SNAPSHOT_HARNESS_GENERATOR_HELP                                 \
   "memory snapshot harness generator (--harness-type\n"                        \
-  "  initialise-from-memory-snapshot)\n\n"                                     \
+  "  initialise-with-memory-snapshot)\n\n"                                     \
```
🇺🇸

peterschrammel reviewed (approved):
For some reason Github thinks .1 files are not part of the diff and refuses inline comments.
`First,function-call` -> `First, function-call`
initialise 🇺🇸

tautschnig committed:
American spelling: initialise-with-memory-snapshot -> initialize-with-memory-snapshot

All documentation and code has been updated to change the name of the
goto-harness command-line option.
